server/app/services/barman-service.js

```diff
@@ -8,10 +10,118 @@ const { Barman } = require('../models/barman');
  */
 async function getAllBarmen() {
 
-    logger.info('Barman service: get all barmen');
+    logger.verbose('Barman service: get all barmen');
     return await Barman.findAll();
 }
 
+/**
+ * Create a Barman.
+ *
+ * @param newBarman {Barman} partial member
+ * @return {Promise<Barman|Errors.ValidationError>} The created barman with its id
+ */
+async function createBarman(newBarman) {
+
+    logger.verbose('Barman service: creating a new barman named %s %s', newBarman.firstName, newBarman.lastName);
+    return await newBarman.save();
+}
+
+/**
+ * Get a Barman by his id.
+ * @param barmanId {number} Barman Id
+ * @returns {Promise<Barman>} Barmen
+ */
+async function getBarmanById(barmanId) {
+
+    logger.verbose('Barman service: get a barman by his id %d', barmanId);
+
+    const barman = Barman.findById(barmanId);
+
+    if (!barman) throw createUserError('UnknownBarman', 'This Barman does not exist');
+
+    return await barman;
+}
+
+/**
+ * Update a barman
+ * This will copy only the allowed changes from the `updatedBarman`
+ * into the current barman.
+ * This means, with this function, you can not change everything like
+ * the `createdAt` field or others.
+ *
+ * @param barmanId {number} barman id
+ * @param updatedBarman {Barman} Updated barman, constructed from the request.
+ * @return {Promise<Barman>} The updated barman
+ */
+async function updateBarmanById(barmanId, updatedBarman) {
+
+    const currentBarman = await Barman.findById(barmanId);
+
+    if (!currentBarman) throw createUserError('UnknownBarman', 'This Barman does not exist');
+
+    logger.verbose('Barman service: updating barman named %s %s', currentBarman.firstName, currentBarman.lastName);
+
+
+    return await currentBarman.update({
+        firstName: updatedBarman.firstName,
+        lastName: updatedBarman.lastName,
+        nickname: updatedBarman.nickname,
+        facebook: updatedBarman.facebook,
+        godFather: updatedBarman.godFather,
+        dateOfBirth: updatedBarman.dateOfBirth,
+        flow: updatedBarman.flow,
+        kommissions: updatedBarman.kommissions,
```

Comment:
Do not update the association fields for now :) So remove kommissions, roles & services line